pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go

```diff
@@ -132,9 +149,26 @@ func (c *SyncController) SetupWithManager(mgr controllerruntime.Manager) error {
 		},
 	})
 
+	frqPredicate := builder.WithPredicates(predicate.Funcs{
+		// TODO(zhzhuang-zju) To maintain compatibility during upgrades, filtering for work creation events will not be added for now. Once version v1.15 is released, we will be able to exclude works where obj.Spec.StaticAssignments is nil.
+		CreateFunc: func(event.CreateEvent) bool {
+			return true
+		},
+		UpdateFunc: func(updateEvent event.UpdateEvent) bool {
+			oldObj := updateEvent.ObjectOld.(*policyv1alpha1.FederatedResourceQuota)
+			newObj := updateEvent.ObjectNew.(*policyv1alpha1.FederatedResourceQuota)
+			return !reflect.DeepEqual(oldObj.Spec.StaticAssignments, newObj.Spec.StaticAssignments)
+		},
+		// TODO(zhzhuang-zju) To maintain compatibility during upgrades, filtering for work deletion events will not be added for now. Once version v1.15 is released, we will be able to exclude works where obj.Spec.StaticAssignments is nil.
```

Comment:
```suggestion
		// TODO(zhzhuang-zju): Should ignore the delete event in case of the FRQ's .spec.staticAssignments is nil, as
		// no longer sync empty ResourceQuotas by default starting from v1.14. Here temporarily omitting this filter to
		// ensure that during upgrades from older versions, there is an opportunity to clean up legacy ResourceQuotas synced
		// from previous versions.
```